js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+---
+---
+# The GraphQL Transform
+
+The GraphQL Transform is a library that simplifies the process of designing GraphQL APIs.
+In short, it allows a developer to design application backends with a simple `schema.graphql`
+which is then *transformed* into a AWS CloudFormation that implements the data model
+defined in `schema.graphql`. For example you might create the backend for a blog like this:
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
```

Comment:
This is just a sample as there are more annotations such as `@auth` and `@searchable` defined below. You can also write your own transformation to meet your custom business needs.